### ISSUE TITLE
Add an interface for creating editig, assigning  and deleting, roles and permissions

### ISSUE THREAD

**Issue by kamilmedia** [OWASP/rbac#72]
Hi All
I'm a very beginner in PHP RBAC, and I'd like to see in the next release an interface (GUI) to add, edit, delete, roles and permissions, and also assigning roles to a user's table that I choose in config.php, and maybe display a hierarchy af roles and permissions, because  the demo does'nt work for other users just for root.
Thank you in advance

Best regards
Zak

**Comment by abiusx** [comment]
Hi Zak,
Thanks for the feedback.
That actually is planned for the next release!

---

Notice: This message is digitally signed, its source and integrity are verifiable.
If you mail client does not support S/MIME verification, it will display a file (smime.p7s), which includes the X.509 certificate and the signature body.  Read more at Certified E-Mail with Comodo and Thunderbird http://abiusx.com/certified-e-mail-with-comodo-and-thunderbird/ in AbiusX.com

**Comment by nextglory** [comment]
@abiusx Yes, a visualization to the tree is desired, the lefts and rights driving me crazy..:)

**Comment by derBoehm** [comment]
@abiusx can you upload a simple Interface?

**Comment by abiusx** [comment]
A simple interface for what?

**Comment by derBoehm** [comment]
create roles (add permissions to roles) with a HTML GUI.

[http://www.sawmill.net/sawmilltmp/picts/docs/english/tech_manual/roles1.png](http://www.sawmill.net/sawmilltmp/picts/docs/english/tech_manual/roles1.png)

**Comment by abiusx** [comment]
Oh that is the next step on our TODO list. Unfortunately I do not have the time at the moment, and other collaborators are not active.
The infrastructure is all there, it just needs addition of a CSS/JS Tree GUI library, plenty of which are available, and testing.

**Comment by derBoehm** [comment]
Can you send/Upload what you have? I don't need the JS/CSS part coz i do it on my own plus i can send it back if you like it you can have it. I think you know your Code better and I would save some time

**Comment by abiusx** [comment]
Unfortunately I do not have any production ready code. The one that I used before, used a <ul><input type=‘check’> to list all roles as a tree, and let you select those who you want to assign.

**Comment by derBoehm** [comment]
hmm well it's a start and i can see how this works.
would be nice if you can share it

**Comment by abiusx** [comment]
This is the the latest I could fine. The code is more than 5 years old, take a peak first to see if its any good!

**Comment by derBoehm** [comment]
thank you but i cant find the code in your answer :)

**Comment by abiusx** [comment]
https://github.com/abiusx/jframework/blob/master/_japp/view/default/rbac/addrole.php https://github.com/abiusx/jframework/blob/master/_japp/view/default/rbac/addrole.php

**Comment by derBoehm** [comment]
perfekt ty 👍
